app/controllers/devise/two_factor_authentication_controller.rb

```diff
@@ -123,17 +132,9 @@ def handle_second_factor_locked_resource
       sign_out
     end
 
-    def prompt_for_otp_reentry
-      if current_otp_method == :totp
-        show_totp_prompt
-      else
-        show_direct_otp_prompt
-      end
-    end
-
     def send_user_otp
       current_user.send_new_otp(otp_method: current_otp_method)
-      show_direct_otp_prompt
+      redirect_to otp_confirm_path(otp_method: current_otp_method)
```

Comment:
Something is not right about this because the first 2 statements in the confirm method will always return false because the current_otp_method can only be voice or sms when `send_user_otp` is called. This is not new to this PR, but I wanted to make a note of it to remember to address it as part of my 2FA refactor.